include/boost/geometry/formulas/point_segment_distance.hpp

```diff
@@ -0,0 +1,729 @@
+// Boost.Geometry
+
+// Copyright (c) 2019 Oracle and/or its affiliates.
+
+// Contributed and/or modified by Vissarion Fysikopoulos, on behalf of Oracle
+
+// Use, modification and distribution is subject to the Boost Software License,
+// Version 1.0. (See accompanying file LICENSE_1_0.txt or copy at
+// http://www.boost.org/LICENSE_1_0.txt)
+
+#ifndef BOOST_GEOMETRY_FORMULAS_POINT_SEGMENT_DISTANCE_HPP
+#define BOOST_GEOMETRY_FORMULAS_POINT_SEGMENT_DISTANCE_HPP
+
+#include <boost/algorithm/minmax.hpp>
+
+#include <boost/geometry/core/access.hpp>
+#include <boost/geometry/core/radian_access.hpp>
+
+#include <boost/geometry/formulas/result_direct.hpp>
+#include <boost/geometry/formulas/mean_radius.hpp>
+#include <boost/geometry/formulas/meridian_inverse.hpp>
+
+#include <boost/geometry/strategies/distance.hpp>
+#include <boost/geometry/strategies/concepts/distance_concept.hpp>
+
+#include <boost/geometry/strategies/spherical/distance_cross_track.hpp>
+#include <boost/geometry/strategies/spherical/distance_haversine.hpp>
+#include <boost/geometry/strategies/spherical/point_in_point.hpp>
+#include <boost/geometry/strategies/geographic/azimuth.hpp>
+#include <boost/geometry/strategies/geographic/distance.hpp>
+#include <boost/geometry/strategies/geographic/parameters.hpp>
+#include <boost/geometry/strategies/geographic/intersection.hpp>
+
+#ifndef BOOST_GEOMETRY_DETAIL_POINT_SEGMENT_DISTANCE_MAX_STEPS
+#define BOOST_GEOMETRY_DETAIL_POINT_SEGMENT_DISTANCE_MAX_STEPS 100
+#endif
+
+namespace boost { namespace geometry { namespace formula
+{
+
+template <
+    typename CT,
+    bool EnableClosestPoint = false,
+    typename FormulaPolicy = strategy::andoyer,
+    bool Bisection = false
+>
+class point_segment_distance
```

Comment:
You mean `geographic_distance`?

Also the order of template parameters is not consistent with the geographic strategies so may be confusing.

Reply:
renamed to `geographic_point_segment_distance` and changed template parameters' order.